File: app/src/main/java/com/example/TibiaTools/View/ViewModel/ViewModelHouses.java
package com.example.TibiaTools.View.ViewModel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.TibiaTools.APISERVER.models.Houses.House;
import com.example.TibiaTools.APISERVER.models.Houses.Houses;
import com.example.TibiaTools.Repository.RepositoryHouses;

import java.util.ArrayList;
import java.util.List;

public class ViewModelHouses extends ViewModel {
    RepositoryHouses repositoryHouses = new RepositoryHouses();
    private final MutableLiveData<Houses> _houses = new MutableLiveData<Houses>();
    public LiveData<Houses> houses() { return _houses; }
    private final MutableLiveData<ArrayList<String>> _worlds = new MutableLiveData<>();
    public MutableLiveData<ArrayList<String>> Worlds() {return _worlds;}

    public ViewModelHouses(){
        repositoryHouses.worlds(_worlds);
    }
    public void setHouses(String world, String town) {
        repositoryHouses.houses(world, town,_houses);
    }
}

File: app/src/main/java/com/example/TibiaTools/View/ViewModel/ViewModelHouses.kt
package com.example.TibiaTools.View.ViewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.TibiaTools.data.model.Houses
import com.example.TibiaTools.domain.usecase.GetHousesInformationUseCase
import com.example.TibiaTools.domain.usecase.GetWorldsUseCase
import kotlinx.coroutines.launch

class ViewModelHouses(
    private val getHousesInformationUseCase: GetHousesInformationUseCase,
    private val getWorldsUseCase: GetWorldsUseCase
) : ViewModel() {
    private val _houses = MutableLiveData<Houses?>()
    fun houses(): LiveData<Houses?> = _houses

    private val _worlds = MutableLiveData<ArrayList<String>?>()
    fun Worlds(): LiveData<ArrayList<String>?> = _worlds

    init {
        loadWorlds()
    }

    private fun loadWorlds() {
        viewModelScope.launch {
            runCatching { getWorldsUseCase() }
                .onSuccess { dataWords ->
                    val arrayWorlds = arrayListOf("Seleccione")
                    dataWords.worlds?.regular_worlds?.forEach { world ->
                        world.name?.let { arrayWorlds.add(it) }
                    }
                    _worlds.value = arrayWorlds
                }
                .onFailure { _worlds.value = null }
        }
    }

    fun setHouses(world: String, town: String) {
        viewModelScope.launch {
            runCatching { getHousesInformationUseCase(world, town) }
                .onSuccess { _houses.value = it.houses }
                .onFailure { _houses.value = null }
        }
    }
}
